src/applications/vaos/utils/eligibility.js

```diff
@@ -40,25 +41,55 @@ export async function getEligibilityData(
   const facilityId = facility.institutionCode;
   const eligibilityChecks = [
     checkPastVisits(systemId, facilityId, typeOfCareId, 'request').catch(
-      handleRequestError,
+      error => {
+        handleRequestError(error);
+        recordEvent({
+          event: 'vaos-error',
+          'error-key': 'request-past-visits-error',
+        });
+      },
     ),
-    getRequestLimits(facilityId, typeOfCareId).catch(handleRequestError),
+    getRequestLimits(facilityId, typeOfCareId).catch(error => {
+      handleRequestError(error);
+      recordEvent({
+        event: 'vaos-error',
+        'error-key': 'request-exceeded-outstanding-requests-error',
+      });
+    }),
   ];
 
   if (facility.directSchedulingSupported && isDirectScheduleEnabled) {
     eligibilityChecks.push(
       checkPastVisits(systemId, facilityId, typeOfCareId, 'direct').catch(
-        handleDirectError,
+        error => {
```

Comment:
I wonder if we could replace the `handleDirectError` and `handleRequestError` wrapper functions with something like this:

```
function createErrorHandler(directOrRequest, key) {
  return (data) => {
    captureError(data);
    recordEvent({
        event: 'vaos-error',
        'error-key': key,
      });
    return { [`${directOrRequest}Failed`]: true };
  };
}
```

Then do

```
checkPastVisits(systemId, facilityId, typeOfCareId, 'direct')
  .catch(createErrorHandler('direct', 'direct-check-past-visits-error'))
```